File: crates/generate/ess-synth/tests/fixtures/accessor-mechanical.go
package system

import (
	"example.invalid/projection/types/core"
	"testing"
)

func TestDeclaredNominalConversion(t *testing.T) {
	body := core.Body{Status: core.NewSourceTag("ready")}
	event := core.Arrived{Data: body, Choice: core.ChoiceGone{Value: core.NewSourceTag("gone")}, Wrapped: core.NewWrapped(core.Wrapper{Body: body})}
	if input := Project(event); input.Text.Value() != "ready" {
		t.Fatal("conversion lost terminal value")
	}
}

File: crates/generate/ess-synth/tests/fixtures/accessor-mechanical.rs
use projection_types::core::{Arrived, Body, Choice, SourceTag, Wrapped, Wrapper};

#[test]
fn declared_nominal_conversion_preserves_the_terminal_value() {
    let body = Body { status: SourceTag("ready".into()), nested: None };
    let event = Arrived {
        data: body.clone(), partial: None, choice: Choice::Gone(SourceTag("gone".into())),
        wrapped: Wrapped(Wrapper { body }),
    };
    let input = projection_system::project(&event);
    assert_eq!(input.text.0, "ready");
}
